drivers/files/file_access_horizon.cpp

```diff
@@ -0,0 +1,329 @@
+/*************************************************************************/
+/*  file_access_unix.cpp                                                 */
+/*************************************************************************/
+/*                         This file is part of:                         */
+/*                          PANDEMONIUM ENGINE                           */
+/*             https://github.com/Relintai/pandemonium_engine            */
+/*************************************************************************/
+/* Copyright (c) 2022-present Péter Magyar.                              */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "file_access_horizon.h"
+
+#include "core/os/os.h"
+#include "core/string/print_string.h"
+
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include <errno.h>
+
+#if defined(UNIX_ENABLED)
```

Comment:
Unnecessary checks for this platform.
- Unix is always defined (Horizon is based on BSD)
- Android is always undefined (Horizon is not Android-based, thankfully)

Reply:
Yeah, I wanted to clean these, but then it took way too long to get my switch to work with homebrew, and I forgot.

However properly cleaning up all of these will take way too long, and I don't have the time for it anymore, as yesterday I got a bunch of unexpected work.

I don't really see any easy way out of this, that can be done without too much messing around, so I think I'll have to close the pr.

Now that I think of it I probably should have just submitted an engine update pr, but I won't have time to do that properly anymore either. 

I'll keep the original dev commits branch though, in case it's needed.

Reply:
As long as you have "Allow maintainers to make changes" ticked, I can fix it for you later.